ember.md

```diff
@@ -0,0 +1,544 @@
+# Ember.js Style Guide
+
+## Table Of Contents
+
+* [General](#general)
+* [Organizing your modules](#organizing-your-modules)
+* [Models](#models)
+* [Controllers](#controllers)
+* [Templates](#templates)
+* [Routing](#routing)
+* [Ember data](#ember-data)
+
+## General
+
+### Import what you use, do not use globals
+
+For Ember Data, we should import `DS` from `ember-data`, and then
+destructure our desired modules.
+For Ember, use destructuring [as Ember's internal organization is
+not intuitive and difficult to grok, and we should wait until Ember has been
+correctly modularized.](https://github.com/ember-cli/ember-cli-shims/issues/53)
+
+Once Ember has officially adopted shims, we will prefer shims over
+destructuring.
+
+```javascript
+// Good
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const { computed } = Ember;
+
+const { attr } = DS;
+
```

Comment:
Oh, I'm sorry. I didn't look at your example code carefully enough.

Actually, I don't think that'll work. In the first section ( https://github.com/ActionSprout/coding-styles/blob/fea5ea91f6893c2ef919cbc7f2eda4c333d6e545/ember.md#import-what-you-use-do-not-use-globals ), there's a reason for this.

> Ember's internal organization is not intuitive and difficult to grok, and we should wait until Ember has been correctly modularized.
>
> Once Ember has officially adopted shims, we will prefer shims over destructuring.

There's a current project to support these shims but the suggestion is to not use it: https://github.com/ember-cli/ember-cli-shims/issues/53

There's a bunch of discussion going on in rfcs on how to get this right:

* https://github.com/emberjs/rfcs/pull/68
* https://github.com/emberjs/rfcs/pull/176